wasm-tests/webtransport-tests/src/lib.rs

```diff
@@ -0,0 +1,435 @@
+use futures::channel::oneshot;
+use futures::{AsyncReadExt, AsyncWriteExt};
+use getrandom::getrandom;
+use libp2p_core::{StreamMuxer, Transport as _};
+use libp2p_identity::{Keypair, PeerId};
+use libp2p_noise as noise;
+use libp2p_webtransport_websys::{Config, Connection, Error, Stream, Transport};
+use multiaddr::{Multiaddr, Protocol};
+use multihash::Multihash;
+use std::future::poll_fn;
+use std::pin::Pin;
+use wasm_bindgen::JsCast;
+use wasm_bindgen_futures::{spawn_local, JsFuture};
+use wasm_bindgen_test::{wasm_bindgen_test, wasm_bindgen_test_configure};
+use web_sys::{window, Response};
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+#[wasm_bindgen_test]
+async fn single_conn_single_stream() {
+    let addr = fetch_server_addr().await;
+    let keypair = Keypair::generate_ed25519();
+
+    let mut transport = Transport::new(Config::new(&keypair));
+
+    let (_peer_id, mut conn) = transport.dial(addr).unwrap().await.unwrap();
```

Comment:
Should we extract this into a helper `new_connection_to_echo_server`?

Reply:
done